infra/terraform/modules/service/batch.tf

```diff
@@ -0,0 +1,134 @@
+    a_fargate = {
+      name_prefix = "batch-app-fargate"
+
+      compute_resources = {
+        type      = "FARGATE"
+        max_vcpus = 4
+
+        security_group_ids = "${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-API-SG"
+        subnet_ids         = ["${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-BATCH-1A", "${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-BATCH-1B", "${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-BATCH-1C"]
+
+        # `tags = {}` here is not applicable for spot
+      }
+    }
+
+    b_fargate_spot = {
+      name_prefix = "batch-app-fargate_spot"
+
+      compute_resources = {
+        type      = "FARGATE_SPOT"
+        max_vcpus = 4
+
+        security_group_ids = "${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-API-SG"
+        subnet_ids         = ["${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-BATCH-1A", "${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-BATCH-1B", "${var.environment} =\"dev\" ? format(\"%s/%s\",\"DEV/APP\",upper(var.environment)) : upper(var.environment)}-OLCS-PRI-BATCH-1C"]
+        # `tags = {}` here is not applicable for spot
+      }
+    }
+  }
+
+  # Job queus and scheduling policies
+  job_queues = {
+    low_priority = {
+      name     = "BatchTestLowPriorityFargate"
+      state    = "ENABLED"
+      priority = 1
+
+      tags = {
+        JobQueue = "Low priority job queue"
+      }
+    }
+
+    high_priority = {
+      name     = "BatchTestHighPriorityFargate"
+      state    = "ENABLED"
+      priority = 99
+
+      fair_share_policy = {
+        compute_reservation = 1
+        share_decay_seconds = 3600
+
+        share_distribution = [{
+          share_identifier = "A1*"
+          weight_factor    = 0.1
+          }, {
+          share_identifier = "A2"
+          weight_factor    = 0.2
+        }]
+      }
+
+      tags = {
+        JobQueue = "High priority job queue"
+      }
+    }
+  }
+
+  job_definitions = {
+    job_configuration = {
+      name                  = "${var.job_definitions["processQueue"]["job_name"]}-job"
+      type                  = "container"
+      propagate_tags        = true
+      platform_capabilities = ["FARGATE", ]
+
+      container_properties = jsonencode({
+        command = var.job_definitions["processQueue"]["command"],
+        image   = var.job_definitions["processQueue"]["image"],
+        fargatePlatformConfiguration = {
+          platformVersion = "LATEST"
+        },
+        resourceRequirements = [
+          { type = "VCPU", value = "1" },
```

Comment:
As vCPU and Memory are intrinsically linked we need to make both into a variable.

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_size